packages/gatsby/src/commands/develop-process.ts

```diff
@@ -232,10 +258,33 @@ module.exports = async (program: IProgram): Promise<void> => {
           (_, { data }): DataLayerResult => data
         ),
       },
-    }).withContext({ program, parentSpan: bootstrapSpan, app })
+    }).withContext({ program, parentSpan: bootstrapSpan, app, firstRun: true })
   )
+
+  const isInterpreter = <T>(
+    actor: Actor<T> | Interpreter<T>
+  ): actor is Interpreter<T> => `machine` in actor
+
+  const listeners = new WeakSet()
   service.onTransition(state => {
-    reporter.verbose(`transition to ${JSON.stringify(state.value)}`)
+    report.verbose(`Transition to ${JSON.stringify(state.value)}`)
+    // eslint-disable-next-line no-unused-expressions
+    service.children?.forEach(child => {
+      // We want to ensure we don't attach a listener to the same
+      // actor. We don't need to worry about detaching the listener
+      // because xstate handles that for us when the actor is stopped.
+
+      if (isInterpreter(child) && !listeners.has(child)) {
+        child.onTransition(substate => {
+          report.verbose(
+            `Transition to ${JSON.stringify(state.value)} > ${JSON.stringify(
+              substate.value
+            )}`
+          )
+        })
```

Comment:
What's the perf impact of this at runtime when `--verbose` is not used?

Reply:
Shouldn't be much, as it's only executed if the state has changed, but I'll wrap it in a NODE_ENV check to be sure.

Reply:
Realised that this isn't a problem, because this is only used in develop!